src/bridge/Bridge.js

```diff
@@ -300,7 +309,9 @@ class Bridge extends React.Component {
                 setNetworkSeedCache= { this.setNetworkSeedCache }
                 // txn
                 setTxnHashCursor={ this.setTxnHashCursor }
-                txnHashCursor={ txnHashCursor } />
+                txnHashCursor={ txnHashCursor }
+                setTxnConfirmations={ this.setTxnConfirmations }
+                txnConfirmations={ txnConfirmations } />
```

Comment:
@c-johnson I continue to feel highly uncomfortable with the dumping of state into `Bridge.js` for the sole purpose of passing it into new/different screens. In a lot of these cases, especially ones like these where only a single, temporary screen needs the data, it seems much more desirable to pass it in via the `pushRoute` data argument.

How would you feel about standardizing on a pattern for that? Would it be possible to do that in such a way that data is always explicitly passed forward, instead of stored in pseudo-global state? I'm trying to steer towards a slightly more functional style here. @jtobin may also have thoughts.

Reply:
In general I think that's a great idea.  I would just try to ensure `routeData` maintains some sort of sane/predictable structure -- I think using a dynamically-typed blob consisting of whatever the downstream component needs doesn't necessarily scale well, for example.

Reply:
A couple things I can think of; 

- We could bundle the routeData with the actual routeCrumb entry, so it's at least coupled to its view and not a free-floating variable:
in `Bridge.js#pushRoute`

      this.setState((state, _) => ({
        routeCrumbs: state.routeCrumbs.push({
          view: symbol,
          data: routeData
        })
      }));

- Adding propType validators on views that take custom route data to shore up the structure somewhat:

in `Bridge.js#pushRoute`
SentTransaction.propTypes = {
  routeData:  PropTypes.shape({
    promptKeyfile: PropTypes.bool
  })
}

I can't think of many other ways to isolate route data from global state, since all `pushRoute` has access to is updating `Bridge.js`'s state variable. But definitely open for other suggestions if you think of anything.